File: QuickSort/Python/QuickSort.py
import random 

def QuickSort(arr):
    if len(arr) == 1:
        return arr
    elif len(arr) == 0:
        return []
    else:
        
        pivot = arr[-1]
        sortedArr = []
        sortedArr.append(pivot)
        for i in range(len(arr) - 1):
            if arr[i] <= pivot:
                sortedArr.append(arr[i])
            else:
                sortedArr.insert(0, arr[i])
        
        firstHalf = QuickSort(sortedArr[:sortedArr.index(pivot)])
        secondHalf = QuickSort(sortedArr[(sortedArr.index(pivot)+1):])
        secondHalf.insert(0, pivot)
        sortedArr = firstHalf + secondHalf
    
    return sortedArr
            
            


a = [12, 3, 5, 6, 23, 2, 9, 13, 8, 4, 7]
b = [10, 9, 8, 7, 6, 5, 4, 3, 2, 1]
c = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
d = []
for i in range(12):
    n = random.randint(1, 30)
    d.append(n)

print(QuickSort(a))
print(QuickSort(b))
print(QuickSort(c))
print(QuickSort(d))

File: QuickSort/C#/QuickSort.cs
using System;
using System.Linq;
using System.Collections.Generic;

namespace QuickSort
{
	class Program 
	{
		public static List<int> QuickSort(List<int> someList)
		{
			
			if(someList.Count == 1)
				return someList;
			
			List<int> retList = new List<int>();
			
			if(someList.Count == 0)
				return retList;
			
			int listSize = someList.Count;
			int pivot = someList[listSize - 1];
			retList.Add(pivot);

			for(int i = 0; i < listSize - 1 ; i++)
				if(someList[i] <= pivot)
					retList.Add(someList[i]);
				else
					retList.Insert(0, someList[i]);
			int indexOfPivot = retList.IndexOf(pivot);	
			List<int> firstHalf = QuickSort(retList.GetRange(0, indexOfPivot));
			List<int> secondHalf = QuickSort(retList.GetRange(indexOfPivot + 1, retList.Count - indexOfPivot - 1));
			secondHalf.Insert(0, pivot);
			
			return firstHalf.Concat(secondHalf).ToList();
		}

		public static void printList(List<int> anotherList)
		{
			foreach(int member in anotherList)
				Console.Write($"{member} ");
			Console.WriteLine();
		}


		public static void Main(string[] args)
		{
			Random rnd = new Random();

			List<int> myListOne = new List<int>{12, 3, 5, 6, 23, 2, 9, 13, 8, 4, 7};
			List<int> myListTwo = new List<int>{1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
			List<int> myListThree = new List<int>{10, 9, 8, 7, 6, 5, 4, 3, 2, 1};
			List<int> myListFour = new List<int>();

			for(int i = 0; i < 30; i++)
				myListFour.Add(rnd.Next(1, 31));

			printList(myListOne);
			printList(myListTwo);
			printList(myListThree);
			printList(myListFour);
			Console.WriteLine("Sorted: ");

			List<int> retList = QuickSort(myListOne);
			printList(retList);
			
 			retList = QuickSort(myListTwo);
			printList(retList);
 
			retList = QuickSort(myListThree);
			printList(retList);
 
			retList = QuickSort(myListFour);
			printList(retList);

		}
	}
}
